react/getting_started_with_react/rendering_techniques.md

```diff
@@ -0,0 +1,299 @@
+### Introduction
+
+Now that we have learned how JSX works and how to write it. This lesson will cover how we can render multiple elements and conditionally render UI in JSX.
+
+### Lesson Overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+*   Render a list of elements/components in JSX
+*   Conditionally render UI
+
+### Rendering a List of Elements in JSX
+
+Let us say we want to create a component that lists multiple animals. We can do so by doing:
+
+~~~javascript
+function App() {
+  return (
+    <div>
+      <h1>Animals: </h1>
+      <ul>
+        <li>Lion</li>
+        <li>Cow</li>
+        <li>Snake</li>
+        <li>Lizard</li>
+      </ul>
+    </div>
+  );
+}
+~~~
+
+It is perfectly acceptable, but what if we want to render more than just four? It can be tedious and long, and most of the time, we will be dealing with a data structure (like a list) rather than hard coding each animal. You have previously learned that we can embed expressions inside JSX with curly braces. So let us do just that:
+
+~~~javascript
+function App() {
+  const animals = ["Lion", "Cow", "Snake", "Lizard"];
+  return (
+    <div>
+      <h1>Animals: </h1>
+      <ul>
+        {animals.map((animal) => {
+          return <li key={animal}>{animal}</li>;
+        })}
+      </ul>
+    </div>
+  );
+}
+~~~
+
+We have defined an array called `animals` which contains what we have previously written. Now inside our JSX, we used `map` to map over each animal in the list and return the `li` element, adding `animal` as its text. It should now render the same as the previous snippet we wrote. It is now more dynamic and easy to handle.
```

Comment:
I feel we should have an explanation on why this code works: 

```js
 {animals.map((animal) => {
          return <li key={animal}>{animal}</li>;
  })}
```

Suggestion:
```suggestion
~~~javascript
function App() {
  const animals = ["Lion", "Cow", "Snake", "Lizard"];
  return (
    <div>
      <h1>Animals: </h1>
      <ul>
        {animals.map((animal) => {
          return <li key={animal}>{animal}</li>;
        })}
      </ul>
    </div>
  );
}
~~~

We define an array called `animals`. Now inside our JSX, we use `map` to return a new array of `li` elements, adding `animal` as its text. It should now render the same as the previous snippet we wrote. This is because JSX has the ability to automatically render arrays. The following code is identical: 

~~~js
function App() {
  const animals = ["Lion", "Cow", "Snake", "Lizard"];
  const animalsList = animals.map((animal) => <li key={animal}>animal</li>)
  
  return (
    <div>
      <h1>Animals: </h1>
      <ul>
        {animalsList}
      </ul>
    </div>
  );
}
~~~

Reply:
This small change really caught the essence of it. Committing this suggestion.